src/core/remote.js

```diff
@@ -1852,8 +1856,15 @@ Remote.prototype.createPathFind = function(options, callback) {
   }
 
   if (callback) {
+    const updateTimeout = setTimeout(() => {
```

Comment:
Actually I've seen situations where server just stops sending updates - so we will wait indefinitely. I think this was the main reason of hangs that the Abiy saw.